### PR TITLE
DP-370 Fix history

### PR DESCRIPTION
**Pull request by Ronserruya**
Fixed paging with txs history, before the fix the "get_tx_history" method returned None if it had to use paging.

Also changed the implementation from recursion to loop

### CODE REVIEW

File: kin/client.py [hunk 1]
```diff
@@ -213,31 +213,33 @@ def kin_account(self, seed: str, channel_secret_keys: Optional[List[str]] = None
...
+                raw_tx = RawTransaction(transaction)
+                if simple:
+                    try:
+                        simple_tx = SimplifiedTransaction(raw_tx)
+                        current_loop_txs.append(simple_tx)
+                    except KinErrors.CantSimplifyError:
+                        pass
+                else:
+                    current_loop_txs.append(raw_tx)
+                cursor = transaction['paging_token']
```
Comment (katzio) [inline kin/client.py]: I don't see a use for `cursor`, guess it's a leftover

File: kin/client.py [hunk 2]
```diff
@@ -213,31 +213,33 @@ def kin_account(self, seed: str, channel_secret_keys: Optional[List[str]] = None
...
+
+            current_loop_txs = []
+            for transaction in horizon_response['_embedded']['records']:
+                raw_tx = RawTransaction(transaction)
+                if simple:
+                    try:
+                        simple_tx = SimplifiedTransaction(raw_tx)
+                        current_loop_txs.append(simple_tx)
+                    except KinErrors.CantSimplifyError:
+                        pass
```
Comment (katzio) [inline kin/client.py]: hm, why not append it as is (like you did in the `else` section)?

File: kin/client.py (hunk 1 above)
Comment (Ronserruya) [reply kin/client.py]: It is used for the next request, line 220

File: kin/client.py (hunk 2 above)
Comment (Ronserruya) [reply kin/client.py]: That's just the way the method works, if you only want simple txs it skips everything else

**Comment by yosriz** [comment]
@Ronserruya can you add tests for the fix? i.e a scenario with paging?

**Comment by Ronserruya** [comment]
> @Ronserruya can you add tests for the fix? i.e a scenario with paging?

There is already a test for paging

**Pull request merged**